force_wfmanager/ui/setup/mco/kpi_specification_model_view.py

```diff
@@ -40,9 +44,17 @@ def _get_label(self):
         """Gets the label from the model object"""
         if self.model.name == '':
             return "KPI"
-        return "KPI: {} ({})".format(self.model.name, self.model.objective)
+        return "KPI: {} ({})".format(
+            self.model.name, self.model.objective
+        )
 
     #: Listeners
-    @on_trait_change('model.[name,objective]')
-    def kpi_model_change(self):
-        self.model_change()
+    @on_trait_change('model.name,'
+                     'selected_variable.name')
+    def selected_variable_change(self):
```

Comment:
This is an extra check, since the user should not be able to edit the `model.name` through the view anyway